tests/test_config.py

```diff
@@ -2,23 +2,63 @@
 """Functions for testing the global Automata configuration."""
 
 import unittest
-from unittest.mock import patch
+from unittest.mock import patch, MagicMock
+from frozendict import frozendict
 
 import automata.base.config as global_config
+import automata.base.utils
 from automata.fa.dfa import DFA
 
 
 class TestConfig(unittest.TestCase):
 
     def setUp(self):
         self.orig_should_validate = global_config.should_validate_automata
+        self.orig_allow_mutable_automata = global_config.allow_mutable_automata
 
     def tearDown(self):
         global_config.should_validate_automata = self.orig_should_validate
+        global_config.allow_mutable_automata = self.orig_allow_mutable_automata
 
     @patch('automata.fa.dfa.DFA.validate')
     def test_disable_validation(self, validate):
         """Should disable automaton validation"""
         global_config.should_validate_automata = False
         DFA.universal_language({0, 1})
         validate.assert_not_called()
+
+    @patch('automata.base.utils.freezeValue')
+    def test_disable_ensure_values_are_frozen(self, validate):
+        """Should disable the call to freezeValue"""
+        global_config.allow_mutable_automata = True
+        dfa = DFA(
+            states=frozenset(['s1']),
+            input_symbols=frozenset('a'),
+            transitions=frozendict({'s1': frozendict({'a': 's1'})}),
+            initial_state='s1',
+            final_states=frozenset(['s1']),
+        )
+        validate.assert_not_called()
+
+        # Also this should not call ensure freeze nor throw any error
+        dfa = DFA(
+            states={'s1'},
+            input_symbols={'a'},
+            transitions={'s1': {'a': 's1'}},
+            initial_state='s1',
+            final_states={'s1'}
+        )
+        validate.assert_not_called()
+
+    def test_values_are_frozen(self):
```

Comment:
@EduardoGoulart1 I already have tests for every automaton type ([example here](https://github.com/caleb531/automata/blob/main/tests/test_dfa.py#L54)) to test the frozenness (is that a word? 😅 ) of the instance attributes.

Also, one piece of constructive criticism—because you are mocking `freezeValue`, this test will (incorrectly) pass is `freezeValue` is called but not implemented correctly.

All that to say, we should not actually need this test.